libimage/pull.go

```diff
@@ -201,15 +216,16 @@ func (r *Runtime) copyFromDefault(ctx context.Context, ref types.ImageReference,
 		if err != nil {
 			return nil, err
 		}
-		// if index.json has no reference name, compute the image ID instead
-		if manifestDescriptor.Annotations == nil || manifestDescriptor.Annotations["org.opencontainers.image.ref.name"] == "" {
+		storageName = nameFromAnnotations(manifestDescriptor.Annotations)
+		switch len(storageName) {
```

Comment:
Why `switch`?

(BTW `len(s) == 0` and `s == ""` generate exactly the same code, so I’d lean towards the more readable string comparison.)

Reply:
I have no preference and probably flip-flop between the two depending on the mood